tests/snappi_tests/multidut/ecn/test_multidut_dequeue_ecn_with_snappi.py

```diff
@@ -0,0 +1,133 @@
+from tests.snappi_tests.files.helper import skip_ecn_tests
+from tests.common.snappi_tests.common_helpers import packet_capture
+from tests.common.config_reload import config_reload
+from tests.common.snappi_tests.snappi_test_params import SnappiTestParams
+
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [pytest.mark.topology('multidut-tgen')]
+
+
+@pytest.mark.parametrize('line_card_choice', [line_card_choice])
+@pytest.mark.parametrize('linecard_configuration_set', [config_set])
+def test_dequeue_ecn(request,
+                     snappi_api,                    # noqa: F811
+                     conn_graph_facts,              # noqa: F811
+                     fanout_graph_facts,            # noqa: F811
+                     duthosts,
+                     rand_one_dut_lossless_prio,
+                     line_card_choice,
+                     linecard_configuration_set,
+                     get_multidut_snappi_ports,      # noqa: F811
+                     prio_dscp_map):                # noqa: F811
+    """
+    Test if the device under test (DUT) performs ECN marking at the egress
+
+    Args:
+        request (pytest fixture): pytest request object
+        snappi_api (pytest fixture): SNAPPI session
+        conn_graph_facts (pytest fixture): connection graph
+        fanout_graph_facts (pytest fixture): fanout graph
+        duthosts (pytest fixture): list of DUTs
+        rand_one_dut_lossless_prio (str): name of lossless priority to test, e.g., 's6100-1|3'
+        line_card_choice: Line card choice to be mentioned in the variable.py file
+        linecard_configuration_set : Line card classification, (min 1 or max 2  hostnames and asics to be given)
+        prio_dscp_map (pytest fixture): priority vs. DSCP map (key = priority).
+
+    Returns:
+        N/A
+    """
+
+    if line_card_choice not in linecard_configuration_set.keys():
+        assert False, "Invalid line_card_choice value passed in parameter"
+
+    if (len(linecard_configuration_set[line_card_choice]['hostname']) == 2):
+        dut_list = random.sample(duthosts, 2)
+        duthost1, duthost2 = dut_list
+    elif (len(linecard_configuration_set[line_card_choice]['hostname']) == 1):
+        dut_list = [dut for dut in duthosts if
+                    linecard_configuration_set[line_card_choice]['hostname'] == [dut.hostname]]
+        duthost1, duthost2 = dut_list[0], dut_list[0]
+    else:
+        assert False, "Hostname can't be an empty list"
+
+    snappi_port_list = get_multidut_snappi_ports(line_card_choice=line_card_choice,
+                                                 line_card_info=linecard_configuration_set[line_card_choice])
+    if len(snappi_port_list) < 2:
+        assert False, "Need Minimum of 2 ports for the test"
+
+    snappi_ports = get_multidut_tgen_peer_port_set(line_card_choice, snappi_port_list, config_set, 2)
+
+    snappi_port_list = get_multidut_snappi_ports(line_card_choice=line_card_choice,
```

Comment:
why are there two `snappi_port_list` here? L70 and L77

Reply:
Fixed